src/main/java/seedu/address/logic/util/LessonModificationUtil.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.logic.util;
+
+import static seedu.address.logic.commands.AddAttendanceCommand.MESSAGE_DUPLICATE_ATTENDANCE;
+import static seedu.address.logic.commands.DeleteAttendanceCommand.MESSAGE_MISSING_ATTENDANCE;
+
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.attendance.Attendance;
+import seedu.address.model.attendance.AttendanceRecord;
+import seedu.address.model.attendance.AttendanceRecordList;
+import seedu.address.model.attendance.Week;
+import seedu.address.model.lesson.Day;
+import seedu.address.model.lesson.Lesson;
+import seedu.address.model.lesson.NumberOfOccurrences;
+import seedu.address.model.lesson.Venue;
+import seedu.address.model.student.Student;
+
+/**
+ * Contains utility methods for modifying {@code Attendances} in {@code Lessons}.
+ */
+public class LessonModificationUtil {
+
+    /**
+     * Adds all attendances in the target lesson to the new lesson.
+     * The {@code attendanceToAdd} is added to the lesson.
+     *
+     * @throws CommandException if the {@code attendanceToAdd} already exists.
+     */
+    public static Lesson createModifiedLessonWithAddedAttendance(
+            Lesson targetLesson, Student targetStudent, Week targetWeek, Attendance attendanceToAdd)
+            throws CommandException {
+        assert targetLesson != null;
+        assert targetStudent != null;
+        assert targetWeek != null;
+        assert attendanceToAdd != null;
```

Comment:
Should this be a `requireAllNonNull` check, since it is a public method now?
```suggestion
        requireAllNonNull(targetLesson, targetStudent, targetWeek, attendanceToAdd);
```